src/Cohesity.Powershell/Scripts/ProtectionSource/Register-CohesityProtectionSourceIsilon.ps1

```diff
@@ -0,0 +1,73 @@
+function Register-CohesityProtectionSourceIsilon {
+  <#
+        .SYNOPSIS
+        Register a new Isilon source.
+        .DESCRIPTION
+        The Register-CohesityProtectionSourceIsilon function is used to register a new Isilon protection source.
+        .NOTES
+        Published by Cohesity
+        .LINK
+        https://cohesity.github.io/cohesity-powershell-module/#/README
+        .EXAMPLE
+        Register-CohesityProtectionSourceIsilon -Server <string>  -Credential (New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList "root", (ConvertTo-SecureString -AsPlainText "secret" -Force))
+    #>
+  [CmdletBinding()]
+  Param(
+    [Parameter(Position = 0, Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Server,
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNull()]
+    [System.Management.Automation.PSCredential]
+    [System.Management.Automation.Credential()]
+    $Credential
+  )
+
+  Begin {
+    if (-not (Test-Path -Path "$HOME/.cohesity")) {
+      throw "Failed to authenticate. Please connect to the Cohesity Cluster using 'Connect-CohesityCluster'"
+    }
+    $cohesitySession = Get-Content -Path $HOME/.cohesity | ConvertFrom-Json
+    $cohesityCluster = $cohesitySession.ClusterUri
+    $cohesityToken = $cohesitySession.Accesstoken.Accesstoken
+  }
+
+  Process {
+    # Using a private API for the registration, public API will be used in the upcoming release
+    $cohesityClusterURL = $cohesityCluster + '/irisservices/api/v1/backupsources'
+    $cohesityHeaders = @{'Authorization' = 'Bearer ' + $cohesityToken }
+
+    $userName = $Credential.UserName
+    $plainPassword = $Credential.GetNetworkCredential().Password
+    $payload = @{
+      entity     = @{
+        type         = 14
```

Comment:
Instead of declaring direct type value, we can go for enum declaration. As the user won't be aware of the what the number value indicates exactly.

Reply:
Since this is a private API this is what we need to I believe. 

Make it a constant though ISILON_TYPE = 14

Reply:
Fixed